### PR TITLE
Fixes broken specs.

### PR DESCRIPTION
**Pull request by jmangt**
- Updated the version of pdk
- Fixed bash commands to allow directories with spaces in their names
- All specs are now passing

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,4 +2,4 @@ source "https://rubygems.org"
 
 gemspec
 
-gem 'public_suffix', '< 3' if RUBY_VERSION < '2.1'
+gem 'public_suffix', '< 3' if RUBY_VERSION < '2.1'
```
Comment (ekohl) [inline Gemfile]: Please keep the newline at the end

File: puppet-blacksmith.gemspec [hunk 2]
```diff
@@ -18,9 +18,10 @@ Gem::Specification.new do |s|
   s.required_ruby_version = '>= 2.4.0'
 
   s.add_runtime_dependency 'rest-client', '~>2.0'
+  s.add_development_dependency 'bundler'
   s.add_development_dependency 'rake'
-  s.add_development_dependency 'pdk'
-  s.add_development_dependency 'puppet', '>=2.7.16'
+  s.add_development_dependency 'pdk', '~>1.17.0'
```
Comment (ekohl) [inline puppet-blacksmith.gemspec]: Why does it need to be pinned to 1.17.x?

File: .gitignore [hunk 3]
```diff
@@ -3,3 +3,8 @@ pkg/
 tmp/
 .bundle/
 .rspec
+*.tar.gz
+.ruby-gemset
+.ruby-version
+*.code-workspace
+.rspec
```
Comment (ekohl) [inline .gitignore]: This line is a duplicate of line 5

File: puppet-blacksmith.gemspec [hunk 4]
```diff
@@ -18,9 +18,10 @@ Gem::Specification.new do |s|
   s.required_ruby_version = '>= 2.4.0'
 
   s.add_runtime_dependency 'rest-client', '~>2.0'
+  s.add_development_dependency 'bundler'
   s.add_development_dependency 'rake'
-  s.add_development_dependency 'pdk'
-  s.add_development_dependency 'puppet', '>=2.7.16'
```
Comment (ekohl) [inline puppet-blacksmith.gemspec]: Why drop the lower bound? Shouldn't hurt to have one.

File: .gitignore (hunk 3 above)
Comment (jmangt) [reply .gitignore]: fixed

File: puppet-blacksmith.gemspec (hunk 2 above)
Comment (jmangt) [reply puppet-blacksmith.gemspec]: This was the root cause of the specs failing. Unless you forced the dependencies to resolve to the latest pdk. The puppet_spec_helper gem was unable to find pdk. You can see an example in this build https://travis-ci.com/jmangt/puppet-blacksmith/jobs/295415528 
```   
+rake aborted!

      +Errno::ENOENT: No such file or directory - pdk

      +/home/travis/build/jmangt/puppet-blacksmith/vendor/bundle/ruby/2.6.0/gems/puppetlabs_spec_helper-2.14.1/lib/puppetlabs_spec_helper/rake_tasks.rb:150:in ``'

      +/home/travis/build/jmangt/puppet-blacksmith/vendor/bundle/ruby/2.6.0/gems/puppetlabs_spec_helper-2.14.1/lib/puppetlabs_spec_helper/rake_tasks.rb:150:in `rescue in block (2 levels) in <top (required)>'

      +/home/travis/build/jmangt/puppet-blacksmith/vendor/bundle/ruby/2.6.0/gems/puppetlabs_spec_helper-2.14.1/lib/puppetlabs_spec_helper/rake_tasks.rb:144:in `block (2 levels) in <top (required)>'

      +/home/travis/build/jmangt/puppet-blacksmith/vendor/bundle/ruby/2.6.0/gems/puppetlabs_spec_helper-2.14.1/lib/puppetlabs_spec_helper/rake_tasks.rb:129:in `block in <top (required)>'

      +/home/travis/build/jmangt/puppet-blacksmith/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/exe/rake:27:in `<top (required)>'

      +/home/travis/build/jmangt/puppet-blacksmith/vendor/bundle/ruby/2.6.0/bin/ruby_executable_hooks:24:in `eval'

      +/home/travis/build/jmangt/puppet-blacksmith/vendor/bundle/ruby/2.6.0/bin/ruby_executable_hooks:24:in `<main>'

      +

      +Caused by:

      +LoadError: cannot load such file -- pdk/module/build
```

File: puppet-blacksmith.gemspec (hunk 4 above)
Comment (jmangt) [reply puppet-blacksmith.gemspec]: rolled back

File: puppet-blacksmith.gemspec (hunk 2 above)
Comment (ekohl) [reply puppet-blacksmith.gemspec]: Interesting, but I still think it can be pinned to `~> 1.17` so we're not stuck on the 1.17.x line when 1.18 comes out.

**Pull request merged**